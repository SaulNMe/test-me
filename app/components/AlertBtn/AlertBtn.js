import React, { PureComponent } from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import styles from './AlertBtnStyle';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-native/app/styles';



export default class AlertBtn extends PureComponent {
	render() {
		let badgeNumber = (this.props.alertsAmount >= 99) ? '99+' : this.props.alertsAmount;
		let iconBadge = !!(this.props.alertsAmount) ?  
			<View style={styles.iconBadge}>
		    	<Text style={styles.alertNum}>{badgeNumber}</Text>
			</View> : null;
		return (
			<TouchableOpacity
				style={styles.iconContainer}
				onPress={this.props.onPress}
				activeOpacity={0.6}
			>
				<Feather  name='bell' size={24} color={Colors[this.props.iconColor]}/>
				{iconBadge}
			</TouchableOpacity>
		);
	}
}

	AlertBtn.propTypes = {
		alertsAmount: PropTypes.number,
		iconColor: PropTypes.string,
		onPress: PropTypes.func
	}

	AlertBtn.defaultProps = {
		alertsAmount: 0,
		iconColor: 'white',
		onPress: ()=>{}
	}
