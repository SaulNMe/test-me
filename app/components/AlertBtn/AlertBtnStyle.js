import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.flexBox,
	...ApplicationStyles.margins,
	alertNum: {
		color: Colors.white, 
		fontSize: Fonts.size.tiniest,
		fontWeight: Fonts.weight.light
	},
	iconContainer: {
		width: 25,
		height: 25
	},
	iconBadge: {
		height: 17,
		width: 17, 
		borderRadius: 999,
		paddingHorizontal: 2,
		position: 'absolute',
		left: 10,
		top: -5,
		alignItems: 'center', 
		justifyContent: 'center',
		backgroundColor: Colors.red
	},
});
