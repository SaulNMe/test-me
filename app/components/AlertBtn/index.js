import AlertBtn from './AlertBtn';

export default AlertBtn;
