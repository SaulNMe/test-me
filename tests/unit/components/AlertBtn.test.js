import React from 'react';
import { shallow } from 'enzyme';

import AlertBtn from 'qualificame-native/app/components/AlertBtn';
import { Text, TouchableOpacity } from 'react-native';

function createTestProps (props) {
	return {
		// common props
		alertsAmount: 0,
		iconColor: 'white',
		onPress: ()=>{},
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<AlertBtn {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	describe('With default props', () => {
		beforeEach(() => {
			wrapper = createWrapper();
		});
		it('alertsAmount is set to 0', () => {
			let props = wrapper.instance().props;
			expect(props.alertsAmount).toBe(0);
		});
	});
	describe('when alertsAmount is between 0 and 99', () => {
		beforeEach(() => {
			const props = createTestProps({ alertsAmount: 33 })
			wrapper = createWrapper(props);
		});
		it('the badge is rendered', () => {
			expect(wrapper.find(Text).exists()).toBe(true);
			expect(wrapper.find(Text).children().text()).toEqual('33');
		});
	});
	describe('when alertsAmount is bigger than 99', () => {
		beforeEach(() => {
			const props = createTestProps({ alertsAmount: 100 })
			wrapper = createWrapper(props);
		});
		it('the badge is rendered and displays 99+', () => {
			expect(wrapper.find(Text).exists()).toBe(true);
			expect(wrapper.find(Text).children().text()).toEqual('99+');
		});
	});
})

describe('interactions', () => {
	let wrapper;
	const createWrapper = props => shallow(<AlertBtn {...props}/>);
	describe('when onPress function is passed to props', () => {
		let fn;
		beforeEach(() => {
			fn = jest.fn();
			const props = createTestProps({ onPress: fn })
			wrapper = createWrapper(props);
		});
		it('it is called on button press', () => {
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(fn).toBeCalled();
		});
	});
})
