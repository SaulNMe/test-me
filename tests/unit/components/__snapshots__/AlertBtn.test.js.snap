// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.6}
  onPress={[Function]}
  style={
    Object {
      "height": 25,
      "width": 25,
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="#fff"
    name="bell"
    size={24}
  />
</TouchableOpacity>
`;
